### PR TITLE
Copy Windows file metadata

### PR DESCRIPTION
**Pull request by gabriel-samfira**
This change ensures that file ownership and Discretionary ACLs are properly preserved on Windows.

### CODE REVIEW

File: fs/copy_linux.go [hunk 1]
```diff
@@ -27,7 +27,12 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-func copyFileInfo(fi os.FileInfo, name string) error {
```
Comment (dmcgowan) [inline fs/copy_linux.go:30]: Why is it necessary to remove `os.FileInfo` from the interface and force an extra stat called? It is not an exported interface, adding the source string is fine if needed for Windows.

Comment (gabriel-samfira) [reply fs/copy_linux.go:30]: Hi Derek,

I was on the fence about this and was hoping for feedback. If you are okay with adding the source string, I'll gladly add it.

Comment (gabriel-samfira) [reply fs/copy_linux.go:30]: Done

**Comment by jsturtevant** [comment]
xref: https://github.com/containerd/containerd/issues/6235

File: fs/copy_windows.go [hunk 2]
```diff
@@ -21,15 +21,56 @@ import (
...
 	}
 
-	// TODO: copy windows specific metadata
+	// Copy file ownership and ACL
+	// We need SeRestorePrivilege and SeTakeOwnershipPrivilege in order
+	// to restore security info on a file, especially if we're trying to
+	// apply security info which includes SIDs not necessarily present on
+	// the host.
+	privileges := []string{winio.SeRestorePrivilege, seTakeOwnershipPrivilege}
+	if err := winio.EnableProcessPrivileges(privileges); err != nil {
```
Comment (jsturtevant) [inline fs/copy_windows.go:44]: Something similar is done during snapshotting:
 https://github.com/containerd/containerd/blob/7758cdc09ad5c6c5e5eecdfacfa639d3cf87bdbf/snapshots/windows/windows.go#L474-L480

**Review by kzys — APPROVED** [review]

**Review by dmcgowan — APPROVED** [review]

**Pull request merged**